src/fast_array_utils/_plugins/numba_sparse.py

```diff
@@ -0,0 +1,237 @@
+# SPDX-License-Identifier: MPL-2.0
```

Comment:
Should definitely link to https://numba.pydata.org/numba-doc/dev/extending/interval-example.html so people can read up a bit on what is going on here